mlflow/utils/search_utils.py

```diff
@@ -1133,6 +1140,55 @@ def model_version_matches(mv):
 
         return [mv for mv in model_versions if model_version_matches(mv)]
 
+    @classmethod
+    def parse_order_by_for_search_model_versions(cls, order_by):
+        token_value, is_ascending = cls._parse_order_by_string(order_by)
+        identifier = SearchExperimentsUtils._get_identifier(
+            token_value.strip(), cls.VALID_ORDER_BY_ATTRIBUTE_KEYS
+        )
+        return identifier["type"], identifier["key"], is_ascending
+
+    @classmethod
+    def _get_sort_key(cls, order_by_list):
+        order_by = []
+        parsed_order_by = map(cls.parse_order_by_for_search_model_versions, order_by_list or [])
+        for type_, key, ascending in parsed_order_by:
+            if type_ == "attribute":
+                # Need to add this mapping because version is a keyword in sql
+                if key == "version_number":
+                    key = "version"
+                order_by.append((key, ascending))
+            else:
+                raise MlflowException.invalid_parameter_value(f"Invalid order_by entity: {type_}")
+
+        # Add a tie-breaker
+        if not any(key == "last_updated_timestamp" for key, _ in order_by):
+            order_by.append(("last_updated_timestamp", False))
+
+        # https://stackoverflow.com/a/56842689
+        class _Reversor:
+            def __init__(self, obj):
+                self.obj = obj
+
+            # Only need < and == are needed for use as a key parameter in the sorted function
+            def __eq__(self, other):
+                return other.obj == self.obj
+
+            def __lt__(self, other):
+                return other.obj < self.obj
+
+        def _apply_reversor(model, key, ascending):
+            attr = getattr(model, key)
+            return attr if ascending else _Reversor(attr)
```

Comment:
I filed https://github.com/mlflow/mlflow/pull/7713 to reuse `_Reversor`. Can you update the code once it's merged?

Reply:
Merged!

Reply:
@serena-ruan Let's rebase on master and reuse `_Reversor`!